docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -0,0 +1,203 @@
+
+    Select **Insert > Add-ins**. Choose **Manage My Add-ins** and select **Upload My Add-in**. Click "Browse..." for your manifest file (`C:\Users\LabUser\Stock Ticker\manifest.xml`), then click Open, select **Upload**. 
+
+    4. Finally, change the script tag to point to the right custom functions source. Open up your add-in project in your favorite code editor. In **index.html** in the root folder, delete and replace the script tag immediately following the <title> tags with the code below:
+
+    ```js
+    <script src="https://unpkg.com/@microsoft/office-js@1.1.9-adhoc.22/dist/custom-functions-runtime.js" type="text/javascript"></script>
+    ```
+
+## Try out a basic computational custom function
+
+Now the custom functions in your file will be loaded and ready to use. There are several pre-built functions for you in the Yo Office project. All are attached to a namespace called CONTOSO which is defined in the XML manifest file. Once you start typing =CONTOSO in a cell, the list of available functions will appear.
+
+Let's call =CONTOSO.ADD42(). This function adds 42 to any two numbers you provide as arguments. In any cell, type =CONTOSO.ADD42(1,2). It should deliver the answer 45.
+
+## Create a custom function
+
+What if you wanted a function which could fetch and display the price of Microsoft stock that day? Custom functions are designed so you can easily make requests for data from the web asynchronously.
+
+You’ll be adding a new function, called =CONTOSO.STOCKPRICE, to the **customfunctions.j** file. The function will take in the name of a stock ticker, such as "MSFT", and return the price of that stock. You'll leverage the IEX Trading API, which is free and does not require authentication.
+
+    1. Open your code editor of choice and navigate to the stock-ticker project folder. 
+    2. Copy and paste the function below and add it to **customfunctions.js**.
+
+    ```js
+    function STOCKPRICE(ticker) {
+        return new Promise(
+            function(resolve) {
+                let xhr = new XMLHttpRequest();
+                let url = "https://api.iextrading.com/1.0/stock/" + ticker + "/price"
+                //add handler for xhr
+                xhr.onreadystatechange = function() {
+                    if (xhr.readyState == XMLHttpRequest.DONE) {
+                    //return result back to Excel
+                    resolve(xhr.responseText);
+                    }
+                }
+                //make request
+                xhr.open('GET', url, true);
+                xhr.send();
+        });
+    }
+    ```
+
+    You'll notice in this code that your asynchronous function returns a JavaScript Promise with the data from the IEX Trading API. Asynchronous custom functions require you to either return a new Promise or use JavaScript's async/await syntax.
```

Comment:
Please put remarks like this that are about the code ABOVE the code block. Otherwise, users waste time trying to puzzle out things that you're going to explain anyway because they don't see the explanation until after the code block.